config/internal/configsource/manager.go

```diff
@@ -179,21 +186,22 @@ type Manager struct {
 func NewManager(_ *configparser.Parser) (*Manager, error) {
 	// TODO: Config sources should be extracted for the config itself, need Factories for that.
 
-	return &Manager{
-		// TODO: Temporarily tests should set their config sources per their needs.
-		sessions:   make(map[string]configsource.Session),
-		watchingCh: make(chan struct{}),
-		closeCh:    make(chan struct{}),
-	}, nil
+	// TODO: Temporarily tests should set their config sources per their needs.
+	return newManager(nil), nil
 }
 
-// Resolve inspects the given config.Parser and resolves all config sources referenced
-// in the configuration, returning a config.Parser fully resolved. This must be called only
+// Resolve inspects the given configparser.Parser and resolves all config sources referenced
+// in the configuration, returning a configparser.Parser fully resolved. This must be called only
 // once per lifetime of a Manager object.
 func (m *Manager) Resolve(ctx context.Context, parser *configparser.Parser) (*configparser.Parser, error) {
 	res := configparser.NewParser()
 	allKeys := parser.AllKeys()
 	for _, k := range allKeys {
+		if strings.HasPrefix(k, configSourcesKey) {
+			// Remove everything under config_sources
```

Comment:
Can we clarify why we do this?

Reply:
It treats the `config_sources:` section on the configuration as an addition to the actual configuration: so it reads the `config_sources:` section to create the config sources instances but removes them from the resulting configuration returned by the parser provider. I will enrich this comment.